Speeding up the travis test

glistening opened the pull request:
This patch tries to reduce the travis test duration.
   
The `misc` job is spliited into two jobs:
{ default, minimal-profile } and { no-snapshot }
    
The `coverity` job seems strange because it runs all unit test.
Coverity is static anaylzer, so it should not require runtime test.
The coverity site also says just providing make command is enough.
    
IoT.js-DCO-1.0-Signed-off-by: Sanggyu Lee sg5.lee@samsung.com

LaszloLango commented on .travis.yml:
```diff
@@ -27,6 +27,8 @@ env:
     - OPTS="--test=artik053" INSTALL_TIZENRT_DEPS=yes
     - OPTS="--test=misc"
+    - OPTS="--test=nosnapshot"
+    - OPTS="--test=minimalprofile"
```
I'd write it with dashes to be similar to the build options (`no-snapshot` and `minimal-profile`).

LaszloLango commented:
> The coverity job seems strange because it runs all unit test.
> Coverity is static anaylzer, so it should not require runtime test.
> The coverity site also says just providing make command is enough.

Please check https://github.com/jerryscript-project/jerryscript/blob/master/.travis.yml#L39 and https://github.com/jerryscript-project/jerryscript/blob/master/.travis.yml#L71
This is how we solved this problem in JerryScript.

glistening commented:
@LaszloLango I also prefer the names you suggest. I was not certain that the travis allows dash in name. I will fix it.

For coverity scan, I will refer your link. Thank you.

LaszloLango commented:
@glistening I was thinking a bit more on this coverity issue. I think the correct fix would be this:

```diff
diff --git a/tools/precommit.py b/tools/precommit.py
index 93e7a06..f7047b5 100755
--- a/tools/precommit.py
+++ b/tools/precommit.py
@@ -255,4 +255,4 @@ for test in option.test:
         build("debug", ['--iotjs-minimal-profile'] + build_args)
 
     elif test == "coverity":
-        build("debug", os_dependency_module['linux'] + build_args)
+        build("debug", ['--no-check-test'] + os_dependency_module['linux'] + build_args)
```

glistening commented:
@LaszloLango I've added dash in name, and fixed 80 column rule, and I keep `--no-check-test` option since both you and I seem to be thinking same thing.
